Packs/ZoomMail/Integrations/ZoomMail/ZoomMail.py

```diff
@@ -0,0 +1,1224 @@
+            "license": zoom_license,
+        }
+        return self._http_request(method="GET", url_suffix="/users", params=params)
+
+
+def the_testing_module(client: ZoomMailClient, params: dict) -> str:
+    """
+    Tests authentication for the ZoomMail API by attempting to obtain an access token.
+    """
+    # First we check the parameters are valid
+    err = validate_params(params)
+
+    # Next we test for access
+    token_response = client.obtain_access_token()
+    if "error" in token_response:
+        err.append(token_response.get("error", "Unknown error occurred."))
+
+    # If we have errors, let's give them all at once in a nice neat message
+    if len(err) > 0:
+        return f"Errors were found while testing:\n{''.join(err)}"
+    return "ok"
+
+
+def fetch_incidents(client: ZoomMailClient, params: dict[str, str]) -> None:
+    """
+    Fetches email messages from ZoomMail API and creates incidents.
+
+    :param client: The ZoomMailClient instance.
+    """
+    fetch_from: Optional[str] = params.get("default_mailbox")
+    fetch_query: str = params.get("fetch_query", "")
+    first_fetch_time: str = params.get("first_fetch", "3 days")
+    fetch_labels: str = params.get("fetch_labels", "INBOX")
+
+    max_fetch = min(int(params.get("max_fetch", 50)), 200)
+
+    last_run = demisto.getLastRun()
+    last_fetch_info = last_run.get("last_fetch_info", {"internalDate": 0, "ids": []})
+    last_page_fetch_token = last_run.get("next_page_token")
+    processed_ids: Set[str] = set(last_run.get("processed_ids", []))
+
+    if not last_fetch_info["internalDate"]:
+        first_fetch_dt = parse(first_fetch_time)
+        if not first_fetch_dt:
+            first_fetch_dt = datetime.now() - timedelta(days=3)
+        last_fetch_info["internalDate"] = first_fetch_dt.timestamp()
+
+    new_processed_ids = processed_ids.copy()
+
+    incidents: list[dict[str, Any]] = []
+
+    query = f"{fetch_query} after:{int(last_fetch_info['internalDate'])}"
+    if last_page_fetch_token:
+        demisto.info("Fetching leftover emails from previous fetch run.")
+        messages_response = client.list_emails(
+            email=fetch_from, page_token=last_page_fetch_token, label_ids=fetch_labels
+        )
+    else:
+        messages_response = client.list_emails(
+            email=fetch_from,
+            max_results=str(max_fetch),
+            query=query,
+            label_ids=fetch_labels,
+        )
+    next_page_token = messages_response.get("nextPageToken", None)
+    messages = messages_response.get("messages", [])
+    message_dates: list[float] = []
+
+    for msg in messages:
+        message_id = msg.get("id")
+        thread_id = msg.get("threadId", "")
+        message_details = client.get_email_message(
+            email=fetch_from, message_id=message_id
+        )
+        internal_date = float(message_details.get("internalDate")) / 1000.0
+
+        msg.update({"internalDate": internal_date})
```

Comment:
Because we are looping off the list of fetched emails and then getting the internal time for each one. At the end of the run, I want to sort a list of message IDs and their create time so I can grab all emails at that end time.